pkg/config/labels/instance.go

```diff
@@ -80,7 +80,10 @@ func (i Instance) Equals(that Instance) bool {
 	if that == nil {
 		return i == nil
 	}
-	return i.SubsetOf(that) && that.SubsetOf(i)
+	if len(i) != len(that) {
```

Comment:
could you expand TestLabelsEquals a bit to ensure this doesn't result in any behavioral changes? Right now the tests look pretty light, and I am a bit worried about some edge case